File: Lista de Exercícios/Java/15. Exceções Personalizadas/ExcecoesPersonalizadas.java
class SaldoInsuficienteException extends Exception {
    public SaldoInsuficienteException(String mensagem) {
        super(mensagem);
    }
}

class ContaBancaria {
    private double saldo;

    public ContaBancaria(double saldoInicial) {
        this.saldo = saldoInicial;
    }

    public void sacar(double valor) throws SaldoInsuficienteException {
        if (valor <= 0) {
            throw new IllegalArgumentException("O valor do saque deve ser positivo.");
        }
        if (valor > saldo) {
            throw new SaldoInsuficienteException("Não foi possível realizar o saque devido ao saldo insuficiente.");
        }
        saldo -= valor;
    }

    public void depositar(double valor) {
        if (valor > 0) {
            saldo += valor;
        } else {
            throw new IllegalArgumentException("O valor do depósito deve ser positivo.");
        }
    }

    public double getSaldo() {
        return saldo;
    }
}

File: Lista de Exercícios/Python/15. Exceções Personalizadas/ExcecoesPersonalizadas.py
class SaldoInsuficienteException(Exception):
    def __init__(self, alerta):
        super().__init__(alerta)

class ContaBancaria:
    def __init__(self, saldo = 0):
        self.saldo = saldo

    def sacar(self, valor):
         if 0 < valor <= self.__saldo:
              raise SaldoInsuficienteException("Não foi possível realizar o saque devido ao saldo insuficiente")
